zephyr/CMakeLists.txt

```diff
@@ -900,6 +900,10 @@ zephyr_library_sources_ifdef(CONFIG_SOF_BOOT_TEST
        boot_test.c
 )
 
+zephyr_library_sources_ifdef( CONFIG_TELEMETRY
+       ${SOF_SRC_PATH}/debug/telemetry/telemetry.c
```

Comment:
Do need add C files in this top-level file. CMake and Kconfig belong to subdirectories, I'm trying to scatter this giant file (#8260).

Use Zephyr conditionals in `src/debug/telemetry/CMakeLists.txt` like ef563ebfe3d572 did in `src/init/CMakeLists.txt`

BTW are you testing BOTH Zephyr and non-Zephyr? If you're testing only one then add only the one you tested.

Reply:
I'm testing Zephyr only, building with xtensa-build-zephyr.py. First time adding files here so my first idea was adding CmakeLists in subdirectories but it didn't work for me - it's not even parsed . Should I then use `zephyr/CMakeLists.txt` and delete the subdirectory one or is `zephyr/CMakeLists.txt` still a no-go?

Reply:
Thanks for the additional information. I also took a closer look. Try this:
```diff
--- a/zephyr/CMakeLists.txt
+++ b/zephyr/CMakeLists.txt
@@ -146,6 +146,7 @@ endmacro()
 
 add_subdirectory(../src/init/ init_unused_install/)
 add_subdirectory(../src/ipc/  ipc_unused_install/)
+add_subdirectory(../src/debug/telemetry/  telemetry_unused_install/)
 add_subdirectory(test/)
```

Then add_sources the new C file from its subdirectory _only_. Funny enough you were already doing that but it had no effect yet. This line above should fix that.

So you were adding the same C file from two different places. That's why I wondered whether you were testing the non-Zephyr/XTOS build too.

Generally speaking, don't add CMake code that is already dead! TEST every line of CMake change by temporarily deleting it (much easier and faster than testing C code)

Reply:
Yes, this works, thanks! Check if it looks good now, if you can.